pkg/virtctl/network/dynamicifaces_test.go

```diff
@@ -0,0 +1,146 @@
+package network_test
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakek8sclient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/testing"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/network"
+	"kubevirt.io/kubevirt/tests/clientcmd"
+)
+
+var _ = Describe("Dynamic Interface Attachment", func() {
+	var (
+		ctrl         *gomock.Controller
+		kubeClient   *fakek8sclient.Clientset
+		vmInterface  *kubecli.MockVirtualMachineInterface
+		vmiInterface *kubecli.MockVirtualMachineInstanceInterface
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+
+		kubeClient = fakek8sclient.NewSimpleClientset()
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	const (
+		ifaceName   = "pluggediface1"
+		networkName = "newnet"
+		vmName      = "myvm1"
+	)
+
+	mockVMIAddInterfaceEndpoints := func(vmName string, networkName string, ifaceName string) {
+		kubecli.MockKubevirtClientInstance.
+			EXPECT().
+			VirtualMachineInstance(k8smetav1.NamespaceDefault).
+			Return(vmiInterface).
+			Times(1)
+		vmiInterface.EXPECT().AddInterface(vmName, gomock.Any()).DoAndReturn(func(arg0, arg1 interface{}) interface{} {
+			Expect(arg1.(*v1.AddInterfaceOptions).NetworkName).To(Equal(networkName))
+			Expect(arg1.(*v1.AddInterfaceOptions).InterfaceName).To(Equal(ifaceName))
+			return nil
+		})
+	}
+
+	mockVMAddInterfaceEndpoints := func(vmName string, networkName string, ifaceName string) {
+		kubecli.MockKubevirtClientInstance.
+			EXPECT().
+			VirtualMachine(k8smetav1.NamespaceDefault).
+			Return(vmInterface).
+			Times(1)
+		vmInterface.EXPECT().AddInterface(vmName, gomock.Any()).DoAndReturn(func(arg0, arg1 interface{}) interface{} {
+			Expect(arg1.(*v1.AddInterfaceOptions).NetworkName).To(Equal(networkName))
+			Expect(arg1.(*v1.AddInterfaceOptions).InterfaceName).To(Equal(ifaceName))
+			return nil
+		})
+	}
+
+	DescribeTable("should fail when required input parameters are missing", func(cmdType string, args ...string) {
+		cmd := clientcmd.NewVirtctlCommand(append([]string{cmdType}, args...)...)
+		Expect(cmd.Execute()).To(HaveOccurred())
```

Comment:
`NotTo(Succeed())`

Reply:
> `HaveOccured` is used when directly examining errors and it reads correctly only when that scenario occurs: `Expect(err).To(HaveOccured())`. This is both the convention used and documented.
> 
> When examining directly an operation/method, it does not fit well. Because Execute Have Occurred has no meaning... what exactly occurred? 

... an error ? Let me paste you the docstring of `[HaveOccurred](https://github.com/onsi/gomega/blob/119d4cea3d32068752e4285e256e231d9748757a/matchers.go#L67)`:
```golang
//HaveOccurred succeeds if actual is a non-nil error
//The typical Go error checking pattern looks like:
//    err := SomethingThatMightFail()
//    Expect(err).ShouldNot(HaveOccurred())
```

> When used directly on the method, it sounds correct to say Execute Succeeded and its negative when appropriate.
> 
> If you want to keep `HaveOccured`, use an `err` variable so it will read right.

If this is the price I have to pay, I will pay it; will update it.